### PR TITLE
BENCH-3240: Add exit and error logging to post startup script when importing custom application

### PR DESCRIPTION
**Pull request by june-hua**
Copy over error logging from Vertex AI post startup script to the custom app post startup script.

### CODE REVIEW

File: startupscript/post-startup.sh [hunk 1]
```diff
@@ -73,6 +77,45 @@ ${RUN_AS_LOGIN_USER} "mkdir -p '${USER_BASH_COMPLETION_DIR}'"
...
+# Set guest attributes on GCE. Used here to log completion status of the script.
+# See https://cloud.google.com/compute/docs/metadata/manage-guest-attributes
+# Arguments:
+#   $1: The guest attribute domain and key IE startup_script/status
+#   $2  The data to write to the guest attribute
+#######################################
+function set_guest_attributes() {
+  local attr_path="${1}"
+  local attr_value="${2}"
+  if [ -z "${attr_path}" ] || [ -z "${attr_value}" ]; then
```
Comment (june-hua) [inline startupscript/post-startup.sh:90]: Added some input checking here.

Comment (yuhuyoyo) [reply startupscript/post-startup.sh:90]: try this one https://github.com/verily-src/workbench-app-devcontainers/blob/6af112ee40d05ee3f1d9188cd4076c65de28e285/startupscript/butane/gcp/metadata-utils.sh#L33C10-L33C22

File: startupscript/post-startup.sh [hunk 2]
```diff
@@ -73,6 +77,45 @@ ${RUN_AS_LOGIN_USER} "mkdir -p '${USER_BASH_COMPLETION_DIR}'"
...
+
+# If the script exits without error let the UI know it completed successfully
+# Otherwise if an error occurred write the line and command that failed to guest attributes.
+function exit_handler {
+  local exit_code="${1}"
+  local line_no="${2}"
+  local command="${3}"
+  # Success! Set the guest attributes and exit cleanly
+  if [[ "${exit_code}" -eq 0 ]]; then
+    set_guest_attributes "${STATUS_ATTRIBUTE}" "COMPLETE"
```
Comment (yuhuyoyo) [inline startupscript/post-startup.sh:108]: just do nothing when exit 0. we have another step after this to set complete.

File: startupscript/butane/devcontainer-failure-handler.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
```
Comment (yuhuyoyo) [inline startupscript/butane/devcontainer-failure-handler.sh:2]: set -o errexit
set -o nounset
set -o pipefail
set -o xtrace

can you add these 

and also a short paragraph about what this script do?

File: startupscript/butane/devcontainer-failure-handler.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+readonly RETRY_COUNT=3
+readonly RETRY_FILE="/tmp/devcontainer-failure-count"
+
+# Log failure
+echo "failed: $(date)" >> $RETRY_FILE
+
+# Check if number of retries have exceeded maximum
+num_retries="$(wc -l < $RETRY_FILE)"
+if [ "$num_retries" -ge "$RETRY_COUNT" ]; then
```
Comment (yuhuyoyo) [inline startupscript/butane/devcontainer-failure-handler.sh:11]: [[ "${num_retries}" -ge "${RETRY_COUNT}" ]]

**Review by yuhuyoyo — APPROVED** [review]

**Pull request merged**